source/opcode.cpp

```diff
@@ -106,7 +106,7 @@ spv_result_t spvOpcodeTableNameLookup(spv_target_env env,
     // Note that the second rule assumes the extension enabling this instruction
     // is indeed requested in the SPIR-V code; checking that should be
     // validator's work.
-    if ((static_cast<uint32_t>(env) >= entry.minVersion ||
```

Comment:
How did this work?

Reply:
It just happens that we have target envs for later versions of SPIR-V placed after earlier ones, and client API target envs placed after their required universal SPIR-V versions.

Reply:
But you are not comparing two enums.

On the left you have
static_cast<uint32_t>(SPV_ENV_UNIVERSAL_1_2) = 2

On the right you have it->minVersion which is probably something like 0x10002.

Reply:
Previously the `generate_grammar_tables.py` were written to emit `SPV_ENV_UNIVERSAL_*` as the `minVersion`. So we were comparing enumerant values. This pull request changed that to use `SPV_SPIRV_VERSION_WORD()` everywhere. See https://github.com/KhronosGroup/SPIRV-Tools/pull/1437/files#diff-d07b0ddc717d188e2da6b3342256a34c